tinygrad/codegen/linearizer.py

```diff
@@ -223,82 +229,75 @@ def end_loop(xx:List[Variable]):
     # parse AST
     loaded_buffers = {}
     acc = []
+    val_idxs = global_idxs+local_idxs+upcast_idxs
     self.load_cache: Dict[str, UOp] = {}
     if_gate: Optional[UOp] = None
 
     # reduce op
-    fake_reduce_idxs = []
+    fake_reduce_idxs: List[Variable] = []
     if self.reduceop is not None:
       # define indexes
       reduce_idxs = [Variable(f"ridx{i}", 0, self.full_shape[i]-1) for i in range(self.first_reduce+len(self.group_for_reduce), self.shape_len-self.upcasted)]
       fake_reduce_idxs = [x*0 for x in reduce_idxs]
 
       # define accumulator
       acc = self.global_load(0, global_idxs+local_idxs+fake_reduce_idxs+upcast_idxs, {ReduceOps.SUM: 0.0, ReduceOps.MAX: -math.inf}[cast(ReduceOps, self.reduceop.op)])
+      val_idxs = global_idxs+local_idxs+fake_reduce_idxs+upcast_idxs
+
+      if self.tensor_core:
+        def calc_tc_idxs(local_size: int, aliases: List[List[int]]):
```

Comment:
Does it make sense to define this elsewhere and cache it (with aliases as tuple of tuples)?